astropy/cosmology/parameter/_descriptors.py

```diff
@@ -0,0 +1,47 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import annotations
+
+__all__ = []
+
+from dataclasses import dataclass
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, NoReturn
+
+from astropy.utils.compat.misc import PYTHON_LT_3_10
+
+if TYPE_CHECKING:
+    from astropy.cosmology.core import Cosmology
+
+
+_dataclass_kwargs = {} if PYTHON_LT_3_10 else {"slots": True}
+
+
+@dataclass(frozen=True, **_dataclass_kwargs)
+class ParameterDescriptor:
+    """Immutable mapping of the Parameters.
+
+    If accessed from the class, this returns a mapping of the Parameter
+    objects themselves.  If accessed from an instance, this returns a
+    mapping of the values of the Parameters.
+    """
+
+    attr_name: str = ""
+
+    def __set_name__(self, owner: type[Cosmology], name: str) -> None:
+        object.__setattr__(self, "attr_name", f"_{name}_")
```

Comment:
I haven't seen the "sunder" way do writing private attributes elsewhere. Would suggest to just do `_{name}`

Reply:
SGTM. Doing it now.

Reply:
Done! I commented on this in an above thread, now resolved. Resolving here too :)